### PR TITLE
Changed to pointer receiver  when both were mixed

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Changed to pointer receiver when both were mixed.
- https://go.dev/doc/faq#methods_on_values_or_pointers

### CODE REVIEW

File: cryptoutil/digestset.go [hunk 1]
```diff
@@ -114,9 +114,9 @@ func (first DigestSet) Equal(second DigestSet) bool {
 	return hasMatchingDigest
 }
 
-func (ds DigestSet) ToNameMap() (map[string]string, error) {
+func (first DigestSet) ToNameMap() (map[string]string, error) {
```
Comment (jkjell) [inline cryptoutil/digestset.go:117]: Can you change this to *DigestSet? I still don't have a lot of context on this code but, why the name change to `first`?

File: cryptoutil/digestset.go [hunk 2]
```diff
@@ -190,16 +190,16 @@ func CalculateDigestSetFromFile(path string, hashes []crypto.Hash) (DigestSet, e
 	return CalculateDigestSet(file, hashes)
 }
 
-func (ds DigestSet) MarshalJSON() ([]byte, error) {
-	nameMap, err := ds.ToNameMap()
+func (first DigestSet) MarshalJSON() ([]byte, error) {
```
Comment (jkjell) [inline cryptoutil/digestset.go:193]: Can you change this to `*DigestSet`?

**Review by jkjell — COMMENTED** [review]
Thanks for the PR! Just a few minor things and a question.

File: cryptoutil/digestset.go (hunk 2 above)
Comment (naveensrinivasan) [reply cryptoutil/digestset.go:193]: Marshal doesn't require a `pointer` receiver. https://pkg.go.dev/encoding/json@go1.19#RawMessage.MarshalJSON

I will revert the name change.

File: cryptoutil/digestset.go (hunk 1 above)
Comment (naveensrinivasan) [reply cryptoutil/digestset.go]: 👍

File: cryptoutil/digestset.go (hunk 2 above)
Comment (naveensrinivasan) [reply cryptoutil/digestset.go]: 👍

**Review by jkjell — APPROVED** [review]

**Pull request merged**